src/System.Buffers/src/System/Buffers/DefaultArrayPool.cs

```diff
@@ -20,26 +22,35 @@ internal DefaultArrayPool(int maxLength, int arraysPerBucket)
             if (arraysPerBucket <= 0)
                 throw new ArgumentOutOfRangeException("arraysPerBucket");
 
+            _eventSource = new ArrayPoolEventSource();
```

Comment:
I don't think this is the right design.

There's no need for more than once ArrayPoolEventSource instance.  In fact, more than one getting allocated could cause problems, and the way this is currently set up, more than one could get allocated due to the default array pool being lazily allocated with optimistic concurrency.

The typical pattern is to have a static Log property on the event source, e.g.
```C#
internal sealed class ArrayPoolEventSource : EventSource
{
    internal readonly static ArrayPoolEventSource Log = new ArrayPoolEventSource();
    ...
}
```
and then everything just uses ArrayPoolEventSource.Log to log to it.